src/legacy.js

```diff
@@ -0,0 +1,34 @@
+function migrateLegacyPeriodString(periodString) {
+  if (typeof periodString !== 'string') return periodString;
+
+  const LAST_RANGE_REGEX_LEGACY = /^last(\d+)(day|week|month|quarter|year)s?$/;
+  let match = periodString.match(LAST_RANGE_REGEX_LEGACY);
```

Comment:
Can you use `test` instead of `match` here? Also, always avoid `let`

Reply:
I need match, test will only return a boolean

Reply:
Still I'd rather have `const` and have different variable names rather the overriding the same one over and over, that's a bit weird